src/js/actions/layers.js

```diff
@@ -714,13 +714,16 @@ define(function (require, exports) {
                     }
                 }).then(function () {
                     return this.transfer(tools.resetBorderPolicies);
+                }).then(function () {
+                    return this.transfer(tools.resetPenTool);
```

Comment:
Funny thing is, resetBorderPolicies won't do anything in this bug's case, because it only works on superselect tool. So they can be joined. I'm going to sit on this fix for a while.